source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -1062,12 +1062,27 @@ void Graph::setConstant(UiNodePtr node, mx::InputPtr& input, const mx::UIPropert
             {
                 addNodeInput(_currUiNode, input);
                 input->setValue(temp, input->getType());
-                updateMaterials(input, input->getValue());
+                updateMaterialRequired = true;
             }
         }
     }
 
     ImGui::PopItemWidth();
+
+    if (updateMaterialRequired)
+    {
+        if (!_renderer->getMaterialCompilation() && mx::inputChangeRequiresShaderGen(input))
+        {
+            _frameCount = ImGui::GetFrameCount();
```

Comment:
Use same mechanism other material updates.  Note that we set the flag to not show the ui for "compiling shader" as this clobbers the property editor ui (e.g. the color picker).